### PR TITLE
Added NO_PROXY support for IP address range value

### PR DESCRIPTION
**Pull request by javamachr**
NO_PROXY value can contain ip range values. In such case proxy shouldn't be used but currently is. This PR implements check whether NO_PROXY contains ip value or ip range value such as 192.168.0.1/8 and correctly matches such range against requested URL ignoring proxy for given ranges.

### CODE REVIEW

**Comment by piyush1594** [comment]
ok to test

**Comment by piyush1594** [comment]
Hey, @javamachr Requesting you please add this to CHANGELOG.md

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,13 @@
...
 #### 3.1.11 (To be released)
 
+  Improvements
+     * NO_PROXY setting now supports IP ranges so you can specify whole subnet to be excluded from proxy traffic eg. 192.168.0.1/8
+   
   Bugs
-   * Impersonation parameters not set in withRequestConfig - https://github.com/fabric8io/kubernetes-client/pull/1037
+     * Impersonation parameters not set in withRequestConfig - https://github.com/fabric8io/kubernetes-client/pull/1037
 
+>>>>>>> upstream/master
```
Comment (oscerd) [inline CHANGELOG.md]: Please fix the conflict and squash in one commit.

Comment (javamachr) [reply CHANGELOG.md]: I missed that one sorry, will fix and squash.

Comment (oscerd) [reply CHANGELOG.md]: Thanks man :-)

Comment (javamachr) [reply CHANGELOG.md]: So I managed to squash all my commits into one. I updated my fork from original repo and now PR includes commits from there - not sure if its ok.

Comment (oscerd) [reply CHANGELOG.md]: It doesn't seem to be ok.

**Comment by oscerd** [comment]
ok to test

**Comment by piyush1594** [comment]
Hey @javamachr it will be good if you can go through https://github.com/fabric8io/kubernetes-client/blob/master/CONTRIBUTING.md

**Comment by javamachr** [comment]
I went through it. Next time I'll create a feature branch and an issue and link both. 
I also added 1 more commit with some license reformat to pass licence check.

**Comment by piyush1594** [comment]
Thanks a lot, @javamachr, Looking forward for more contributions :+1:

**Comment by piyush1594** [comment]
Hey, @javamachr Can you please squash the commits one last time, let's get this in ..!!

**Comment by javamachr** [comment]
Squashed once again.

**Comment by piyush1594** [comment]
Thanks a lot @javamachr for your efforts :+1:

**Comment by oscerd** [comment]
ok to test

**Comment by piyush1594** [comment]
[merge]

**Pull request merged**